go/vt/wrangler/traffic_switcher.go

```diff
@@ -1328,20 +1446,39 @@ func (ts *trafficSwitcher) changeRouting(ctx context.Context) error {
 }
 
 func (ts *trafficSwitcher) changeWriteRoute(ctx context.Context) error {
-	rules, err := topotools.GetRoutingRules(ctx, ts.TopoServer())
-	if err != nil {
-		return err
-	}
-	for _, table := range ts.Tables() {
-		delete(rules, ts.TargetKeyspaceName()+"."+table)
-		ts.Logger().Infof("Delete routing: %v", ts.TargetKeyspaceName()+"."+table)
-		rules[table] = []string{ts.TargetKeyspaceName() + "." + table}
-		rules[ts.SourceKeyspaceName()+"."+table] = []string{ts.TargetKeyspaceName() + "." + table}
-		ts.Logger().Infof("Add routing: %v %v", table, ts.SourceKeyspaceName()+"."+table)
-	}
-	if err := topotools.SaveRoutingRules(ctx, ts.TopoServer(), rules); err != nil {
-		return err
+	if ts.isPartialMigration {
+		srr, err := topotools.GetShardRoutingRules(ctx, ts.TopoServer())
+		if err != nil {
+			return err
+		}
+		for _, si := range ts.SourceShards() {
+			delete(srr, fmt.Sprintf("%s.%s", ts.TargetKeyspaceName(), si.ShardName()))
+			ts.Logger().Infof("Delete shard routing: %v:%v", ts.TargetKeyspaceName(), si.ShardName())
```

Comment:
The log messages should be past tense, `Deleted`. Same for the one below: `Added`